corehq/motech/dhis2/entities_helpers.py

```diff
@@ -209,17 +237,113 @@ def register_tracked_entity_instance(requests, case_trigger_info, case_config):
     summaries = response.json()["response"]["importSummaries"]
     if len(summaries) != 1:
         raise Dhis2Exception(_(
-            f'{len(summaries)} tracked entity instances registered from '
-            f'{case_trigger_info}.'
-        ))
+            '{n} tracked entity instances were registered from case {case}.'
+        ).format(n=len(summaries), case=case_trigger_info))
     if case_config["tei_id"] and "case_property" in case_config["tei_id"]:
         case_property = case_config["tei_id"]["case_property"]
         tei_id = summaries[0]["reference"]
         case_updates[case_property] = tei_id
+        # Store `tei_id` in `case_trigger_info` because we might need it
+        # when we create relationships.
+        case_trigger_info.extra_fields[case_property] = tei_id
     if case_updates:
         save_case_updates(requests.domain_name, case_trigger_info.case_id, case_updates)
 
 
+def create_relationships(
+    requests,
+    subcase_trigger_info,
+    subcase_config,
+    dhis2_entity_config,
+):
+    """
+    Creates two relationships in DHIS2; one corresponding to a
+    subcase-to-supercase relationship, and the other corresponding to a
+    supercase-to-subcase relationship.
+    """
+    subcase_tei_id = get_value(subcase_config.tei_id, subcase_trigger_info)
+    if not subcase_tei_id:
+        raise Dhis2Exception(_(
+            'Unable to create DHIS2 relationship: The case {case} is not '
+            'registered in DHIS2.'
+        ).format(case=subcase_trigger_info))
+
+    for rel_config in subcase_config.relationships_to_export:
+        supercase = get_supercase(subcase_trigger_info, rel_config)
+        supercase_config = get_case_config_for_case_type(
+            supercase.type,
+            dhis2_entity_config,
+        )
+        supercase_info = get_case_trigger_info_for_case(
+            supercase,
+            [supercase_config.tei_id],
+        )
+        supercase_tei_id = get_value(supercase_config.tei_id, supercase_info)
+        if not supercase_tei_id:
+            # The problem could be that the relationship is configured
+            # in the CaseConfig for the subcase's case type, but there
+            # is no CaseConfig for the supercase's case type.
+            # Alternatively, registering the supercase in DHIS2 failed.
+            raise ConfigurationError(_(
+                'Unable to create DHIS2 relationship: The case {case} is not '
+                'registered in DHIS2.'
+            ).format(case=supercase_info))
+        if rel_config.supercase_to_subcase_dhis2_id:
+            create_relationship(
+                requests,
+                rel_config.supercase_to_subcase_dhis2_id,
+                supercase_tei_id,
+                subcase_tei_id,
+            )
+        if rel_config.subcase_to_supercase_dhis2_id:
+            create_relationship(
```

Comment:
What happens if above relationship gets created but below one fails for some reason. Should we worry about that case?

Reply:
If DHIS2 returns an error response, it will be caught by the calling function, and reported. It's pretty much what we do with any error we get from DHIS2. The repeat record will be marked as "Failed". (If we fix the configuration, and resend the repeat record, it would change to "Succeeded".)